mantidproject/mantid
# Enable LoadCIF to convert anisotropic to equivalent U

This fixes #15369.

In contrast to the behavior suggested in the list in the issue I decided not to provide a default value in case neither isotropic nor anisotropic parameters are provided, because I'm not sure there is a good default. The value depends on too many things (atomic mass, temperature, crystal structure, ...), so instead the value is set to 0 (basically reverting to the "rigid model").

There used to be one helper class to construct the `CrystalStructure`-object, this is now split into smaller classes, each of which constructs one aspect of the structure. The unit tests for these are separate as well now and some more test cases have been added to check the conversion from anisotropic to isotropic ADPs.

The release notes have been updated (see diff).

There should be a system test with some example structures (created #15539).

**To test:**

You need to have PyCifRW installed for this algorithm to work (as the documentation and errors will tell you). Make sure the doctest still works, it's not executed as part of the build process.

Furthermore it would be best to load some CIFs with anisotropic displacement parameters for some atoms and make sure it produces reasonable results.

You could take the examples for SHELXL2013, CRYSTALS and OLEX2 from [here](http://journals.iucr.org/services/cif/examples.html) and modify the value in the `_atom_site_U_iso_or_equiv` for the first atom to make it invalid (for example prepend a string to the number so that it becomes `a0.052(1)`). When you turn the log level to `information` or `debug`, the algorithm shows a table of atom parameters. The value in the last column should be the same as in the `_atom_site_U_iso_or_equiv` in the CIF file (the number of decimal places may be different, but rounded to the same number of decimal places the values should be the same). 

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
